File: java/src/main/java/com/gushenge/atools/AutoHeightImage.java
package com.gushenge.atools;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.util.AttributeSet;
import androidx.appcompat.widget.AppCompatImageView;

public class AutoHeightImage extends AppCompatImageView {
    public AutoHeightImage(Context context) {
        super(context);
    }

    public AutoHeightImage(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec){
        Drawable d = getDrawable();

        if(d!=null){
            // ceil not round - avoid thin vertical gaps along the left/right edges
            int width = MeasureSpec.getSize(widthMeasureSpec);
            //高度根据使得图片的宽度充满屏幕计算而得
            int height = (int) Math.ceil((float) width * (float) d.getIntrinsicHeight() / (float) d.getIntrinsicWidth());
            setMeasuredDimension(width, height);
        }else{
            super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        }
    }
}

File: kotlin/src/main/java/com/gushenge/atools/ui/AutoHeightImage.kt
package com.gushenge.atools.ui

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView
import kotlin.math.ceil

class AutoHeightImage : AppCompatImageView {
    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val d = drawable
        d?.let {
            // ceil not round - avoid thin vertical gaps along the left/right edges
            val width = MeasureSpec.getSize(widthMeasureSpec)
            //高度根据使得图片的宽度充满屏幕计算而得
            val height = ceil((width.toFloat() * it.intrinsicHeight.toFloat() / it.intrinsicWidth.toFloat()).toDouble()).toInt()
            setMeasuredDimension(width, height)
        }
        d?:super.onMeasure(widthMeasureSpec, heightMeasureSpec)

    }
}
